src/client/Client.cc

```diff
@@ -10018,9 +10018,11 @@ int Client::pwritev(int fd, const struct iovec *iov, int iovcnt, int64_t offset)
 }
 
 int64_t Client::_preadv_pwritev_locked(Fh *fh, const struct iovec *iov,
-				   unsigned iovcnt, int64_t offset, bool write,
-				   bool clamp_to_int, std::unique_lock<ceph::mutex> &cl)
```

Comment:
I'm confused by this change. This lock parameter was introduced in 9d0e88c3ffbec01abffea6ac14f6d27b8a8c1556. It makes sense to me because it makes the lock exception safe.

What's the motivation for changing this?

Reply:
This PR is split from the inode lock patchset in #39797, and with the inode lock it will remove the client_lock and introduce the inode lock instead and will move the inode_lock inside the `_read()/_write()`. 

Just to simplify the inode lock PR to make it easier to review.

Reply:
> It makes sense to me because it makes the lock exception safe.

How is this unsafe? It looks like the old code was just unnecessarily passing in a reference to the client_lock (which is a class member). Is there some case in which client_lock wouldn't resolve to the same thing?

Reply:
It's not normal to directly unlock/lock a mutex held by a locker (e.g. `std::unique_lock`) held by a caller. If an exception is raise, the locker will try to double unlock the mutex during exception unwind. That's why the locker is passed to the callee, so it can use the locker's unlock routine.

Reply:
I'm okay with this interim patch to make #39797 easier.